### ISSUE TITLE
The resolver context includes the authorization header

### ISSUE THREAD

**Issue by sashee** [aws/aws-appsync-community#300]
The ```ctx.request.headers.authorization``` contains the exact authorization header that the client sent in the request.

![image](https://github.com/aws/aws-appsync-community/assets/82075/89109e0d-78f5-4e57-af39-c80ee1b87777)

This is a very sensitive information as that allows anybody who can read it to send requests impersonating the user with that token. And since it's included in the context, it's very easy to accidentally log them:

- A simple console.log leaks it:

```
export function request(ctx) {
  console.log(ctx)
  // ...
}
```

- Passing it to a Lambda, either via the ```payload``` or with a direct lambda resolver and then log the event (very common for debugging):

```
module.exports.handler = async (event) => {
  console.log(event)
  // ...
};
```

I think since the context already contains the ```identity``` with all important information about the caller, the authorization header should be removed or at least truncated.

Interestingly, AppSync resolver logging omits this:

![image](https://github.com/aws/aws-appsync-community/assets/82075/45dc6210-d718-4a70-a20d-b821b2253458)

And at the end of the request, it is truncated:

![image](https://github.com/aws/aws-appsync-community/assets/82075/f1489dea-b545-4537-9faf-330443f31eb2)

**Comment by onlybakam** [comment]
That's correct, when AppSync logs the `context` object, sensitive information is removed from the log message. However, we do not edit/redact any information that is logged via `console.log` functions. The identity data and header data is made available in the context to allow developers to take specific action based on the values
